code/modules/surgery/organs/parasites.dm

```diff
@@ -0,0 +1,35 @@
+/obj/item/organ/internal/body_egg/spider_eggs
+	name = "spider eggs"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "eggs"
+	var/stage = 1
+
+/obj/item/organ/internal/body_egg/spider_eggs/on_life()
+	if(stage < 5 && prob(3))
```

Comment:
Inheritance wasn't quite the right word, what I meant was sibling types from the same basetype that are doing the same thing should do it in the same place unless there's a good reason not to, as it makes life easier for the next person who comes to the code, and reduces the chance of bugs being introduced further down the line.

However, I would agree that it makes more sense to keep the egg in line with the mob's timing rather than the object's timing, it was just when the only reason seemed to be to have it only hatch when the mob was alive, which could have been solved with one small if statement.

Thanks for the explanation :smile: